demo/site/src/layout/header/Header.tsx

```diff
@@ -1,87 +1,248 @@
+                                                            $isExpanded={expandedSubLevelNavigation === node.id}
+                                                        />
+                                                    </ToggleSubLevelNavigationButton>
+                                                )}
+                                            </LinkContainer>
+                                            {node.node.childNodes.length > 0 && (
+                                                <FocusLock disabled={expandedSubLevelNavigation !== node.id} autoFocus={autoFocus}>
+                                                    <SubLevelNavigation id={sublevelMenuId} $isExpanded={expandedSubLevelNavigation === node.id}>
+                                                        <CloseSublevelNavigationButton
+                                                            onClick={() => setExpandedSubLevelNavigation(null)}
+                                                            aria-label={intl.formatMessage(
+                                                                {
+                                                                    id: "header.subMenu.closeButton",
+                                                                    defaultMessage: "Close submenu of {name}",
+                                                                },
+                                                                { name: node.node.name },
+                                                            )}
+                                                            aria-controls={sublevelMenuId}
+                                                        >
+                                                            <SvgUse href="/assets/icons/menu-open.svg#root" width={16} height={16} />
+                                                        </CloseSublevelNavigationButton>
+
+                                                        {node.node.childNodes.map((node) => (
+                                                            <li key={node.id}>
+                                                                <MenuPageLink page={node} activeClassName="active">
+                                                                    {node.name}
+                                                                </MenuPageLink>
+                                                            </li>
+                                                        ))}
+                                                    </SubLevelNavigation>
+                                                </FocusLock>
+                                            )}
+                                        </TopLevelLinkContainer>
+                                    );
+                                })}
+                            </TopLevelNavigation>
+                        </DesktopHeaderFullHeightNav>
```

Comment:
what do you think about moving this also into a child component, similar to MobileMenu?